tests/cnf/ran/diskencryption/tests/tpm2.go

```diff
@@ -0,0 +1,351 @@
+	"github.com/openshift-kni/eco-gotests/tests/cnf/ran/diskencryption/internal/file"
+	stdinmatcher "github.com/openshift-kni/eco-gotests/tests/cnf/ran/diskencryption/internal/stdin-matcher"
+	"github.com/openshift-kni/eco-gotests/tests/cnf/ran/diskencryption/parsehelper"
+	"github.com/openshift-kni/eco-gotests/tests/cnf/ran/internal/cluster"
+	"github.com/openshift-kni/eco-gotests/tests/cnf/ran/internal/raninittools"
+	"github.com/openshift-kni/eco-gotests/tests/internal/inittools"
+)
+
+const (
+	timeoutWaitRegex       = 10 * time.Minute
+	timeoutClusterRecovery = 10 * time.Minute
+)
+
+var _ = Describe("TPM2", Ordered, func() {
+	var (
+		nodeList []*nodes.Builder
+		err      error
+	)
+
+	BeforeAll(func() {
+		if raninittools.BMCClient == nil {
+			Skip("Collecting power usage metrics requires the BMC configuration be set.")
+		}
+	})
+
+	BeforeEach(func() {
+		By("Checks if Secure Boot is enabled")
+		var isSecureBootEnabled bool
+
+		Eventually(func() error {
+			isSecureBootEnabled, err = raninittools.BMCClient.IsSecureBootEnabled()
+
+			return err
+		}).WithTimeout(10 * time.Minute).
+			WithPolling(30 * time.Second).Should(BeNil())
+
+		if !isSecureBootEnabled {
+			By("Enable SecureBoot")
+			Eventually(raninittools.BMCClient.SecureBootEnable).WithTimeout(10 * time.Minute).
+				WithPolling(30 * time.Second).
+				Should(Or(BeNil(), MatchError("secure boot is already enabled")))
+
+			By("Restart node")
+
+			Eventually(raninittools.BMCClient.SystemPowerCycle).WithTimeout(10 * time.Minute).
+				WithPolling(30 * time.Second).Should(BeNil())
+		}
+
+		By("waiting until all spoke 1 pods are ready")
+		err = cluster.WaitForClusterRecover(raninittools.Spoke1APIClient, []string{
+			inittools.GeneralConfig.MCONamespace}, timeoutClusterRecovery)
+
+		if err != nil {
+			By("Server did not come up, maybe the boot order was swapped, swapping back")
+			swapFirstSecondBootItems()
+			By("Restart node")
+			Eventually(raninittools.BMCClient.SystemPowerCycle).WithTimeout(10 * time.Minute).
+				WithPolling(30 * time.Second).Should(BeNil())
+			By("waiting until all spoke 1 pods are ready")
+			err = cluster.WaitForClusterRecover(raninittools.Spoke1APIClient, []string{
+				inittools.GeneralConfig.MCONamespace}, timeoutClusterRecovery)
+			Expect(err).ToNot(HaveOccurred(), "Failed to wait for all spoke 1 pods to be ready")
+		}
+
+		err = file.DeleteFile("/etc/host-hw-Updating.flag")
+		Expect(err).To(Or(MatchError("error deleting file /etc/host-hw-Updating.flag, err=%!w(<nil>)"),
+			MatchError("error deleting file /etc/host-hw-Updating.flag, "+
+				"err=command terminated with exit code 1\nrm: cannot remove '/etc/host-hw-Updating.flag': "+
+				"No such file or directory\r\n")))
+
+		Eventually(func() int {
+			nodeList, err = nodes.List(raninittools.Spoke1APIClient)
+			Expect(err).To(BeNil())
+
+			return len(nodeList)
+		}).Should(BeNumerically("==", 1), "Currently only SNO clusters are supported")
```

Comment:
I need the eventually here because of instability after reboot

Reply:
The sequence in before each was done to support multiple recovery scenarios after testing. The stating point could be a cluster stuck with secureboot disabled and not able to decypt its disk

Reply:
Could you add a comment to explain that please? Most of our other tests won't have that scenario so I think it will be helpful to indicate why this test is different